plugins/Web/WebBackend.vala

```diff
@@ -0,0 +1,176 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2013 Maya Developers (https://launchpad.net/maya)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ * Authored by: Corentin Noël <tintou@mailoo.org>
+ */
+
+public Maya.Backend get_backend (Module module) {
+    debug ("Activating Web Backend");
+    var b = new Maya.WebBackend ();
+    b.ref ();
+    return b;
+}
+
+public static Maya.Backend backend;
+
+public class Maya.WebBackend : GLib.Object, Maya.Backend {
+
+    public WebBackend () {
+        backend = this;
+    }
+
+    public string get_name () {
+        return _("On the web");
+    }
+
+    public string get_uid () {
+        return "webcal-stub";
+    }
+
+    public Gee.Collection<PlacementWidget> get_new_calendar_widget (E.Source? to_edit = null) {
+        var collection = new Gee.LinkedList<PlacementWidget> ();
+
+        bool keep_copy = false;
+        if (to_edit != null) {
+            E.SourceOffline source_offline = (E.SourceOffline)to_edit.get_extension (E.SOURCE_EXTENSION_OFFLINE);
+            keep_copy = source_offline.stay_synchronized;
+        }
+
+        collection.add (Maya.DefaultPlacementWidgets.get_keep_copy (0, keep_copy));
+
+        var url_label = new PlacementWidget ();
+        url_label.widget = new Gtk.Label (_("URL:"));
+        ((Gtk.Label) url_label.widget).expand = true;
+        ((Gtk.Misc) url_label.widget).xalign = 1.0f;
+        url_label.row = 1;
+        url_label.column = 0;
+        url_label.ref_name = "url_label";
+        collection.add (url_label);
+
+        var url_entry = new PlacementWidget ();
+        url_entry.widget = new Gtk.Entry ();
+        ((Gtk.Entry)url_entry.widget).text = "http://";
```

Comment:
Yeah this probably shouldn't be pre-filled. Instead it would be better to set placeholder text:
```suggestion
        url_entry.widget.placeholder_text = "https://example.com";
```

Reply:
I'm seeing now why the cast was probably used. I think you could init with properties to avoid this like:
```
url_entry.widget = new Gtk.Entry () {
    placeholder_text = "https://example.com"
};
```

Reply:
Updated

Reply:
Using a `Granite.ValidatedEntry` might also be a nice touch here but since it's not used anywhere else that can be left to follow up PRs :)